### PR TITLE
[Chimera] Fetching SPM dependencies. 

### PR DESCRIPTION
**Pull request by laxmorek**
### Short description 📝

Working on #1674

This PR introduces support for:
- defining `Swift Package Manager` dependencies in `Dependencies.swift`
- fetching `Swift Package Manager` dependencies using `tuist dependencies fetch` command

### Implementation 👩‍💻👨‍💻

- [x] `ProjectDescription` models - **merged in #2227** 
- [x] `TuistGraph` models & `TuistLoader` mapping - **merged in #2395**
- [x] Fetch and save `SPM` dependencies in `/Tuist/Dependencies` directory **merged in #2656**
- [ ] ~~Build `frameworks`/`xcframeworks`~~
- [x] Acceptance tests **merged in #2656**
- [x] Changelog
- [x] Documentation

**Manifest**

`Dependencies.swift` structure has been changed. Now `platforms` property is top level and shared between dependency managers. 

```swift
import ProjectDescription

let dependencies = Dependencies(
    carthage: .carthage(
        [
            .github(path: "Alamofire/Alamofire", requirement: .exact("5.0.4"))
        ],
        options: [.useXCFrameworks, .noUseBinaries]
    ),
    swiftPackageManager: .swiftPackageManager(
        [
            .package(url: "https://github.com/ReactiveX/RxSwift.git", .exact("5.1.1"))
        ]
    ),
    platforms: [.iOS]
)
```

I can see an opportunity in adding `ExpressibleByArrayLiteral` to `CarthageDependencies` and `SwiftPackageManagerDependencies` to make creation of `Dependencies` more convenient. Im going to add it in one of the upcoming PRs.

### CODE REVIEW

**Comment by pepibumur** [comment]
Could you rebase from main here @laxmorek ?

File: CHANGELOG.md
Comment (danyf90) [inline CHANGELOG.md:9]: Shouldn't it be `SPM` instead of `Carthage`? :D

File: Sources/TuistDependencies/Carthage/CarthageInteractor.swift
Comment (danyf90) [inline Sources/TuistDependencies/Carthage/CarthageInteractor.swift:59]:
```suggestion
    ///   - platforms: List of platforms for which you want to install dependencies.
```

File: CHANGELOG.md
Comment (laxmorek) [reply CHANGELOG.md:9]: right 😅

File: Sources/ProjectDescription/Dependencies/SwiftPackageManagerDependencies.swift
Comment (fortmarek) [inline Sources/ProjectDescription/Dependencies/SwiftPackageManagerDependencies.swift:10]: why is this necessary? I feel like this should be automatically synthesized by the compiler

**Review by fortmarek — APPROVED** [review]
I believe we have already reviewed most of the changes here, LMK @laxmorek if there is something additional to look at since I have just skimmed through it

File: Sources/ProjectDescription/Dependencies/SwiftPackageManagerDependencies.swift
Comment (laxmorek) [reply Sources/ProjectDescription/Dependencies/SwiftPackageManagerDependencies.swift:10]: Right! Im removing `init` from `ProjectDescription.CarthageDependencies` as well.

**Comment by laxmorek** [comment]
> I believe we have already reviewed most of the changes here

Yes, that is true. This PR is a sum of #2227, #2395 and #2656

> LMK @laxmorek if there is something additional to look

What is new: 
- documentation updates: [here](https://github.com/tuist/tuist/pull/2394/files#diff-d3400242174608a4f17807d5e2b1171e7029ed205a5cdf82db8da4cd1f762276)
- new warning log: [here](https://github.com/tuist/tuist/pull/2394/files#diff-285018d6bd9bd7d3d20f5786c50e15c93244d6d9925b5a5d70ac6b0388948309R68)
- update documentation of `Dependencies.init`: [here](https://github.com/tuist/tuist/pull/2394/files#diff-e5177b6881f5c06fdeffa9717ed41a5aacd8870e56de13d24eb467b26908fa06R14) 
- new `CHANGELOG.md` entry: [here](https://github.com/tuist/tuist/pull/2394/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed)

====

@fortmarek let me know if you want to check it.

**Comment by laxmorek** [comment]
After merging `main` into my branch the CI doesn't pass:

Zrzut ekranu 2021-04-9 o 18 30 49

It looks like `/manifests/project` contains broken links. 

=======

In this PR I updated `projects/website/markdown/docs/dependencies/dependencies.mdx` but it's a part of old documentation. I can't find related file in `projects/docs/docs`. 

Is there a plan for migrating this file? How should I handle it in this PR? @pepibumur 

I can create separate PR that moves `dependencies.mdx` to `projects/docs/docs` but if you on it @pepibumur then let me know.

**Comment by fortmarek** [comment]
Thanks @laxmorek for the updates - I have looked over the other changes and looks good 👍 I think it's fine to migrate the documentation in a follow-up PR, so I feel we're good to merge this once CI is green

**Pull request merged**
